cmd/aries-agent-mobile/pkg/wrappers/rest/didexchange.go

```diff
@@ -0,0 +1,122 @@
+	respEnvelope := exec(&restOperation{
+		url:        de.URL,
+		token:      de.Token,
+		httpClient: de.httpClient,
+		endpoint:   de.endpoints[cmddidexch.ReceiveInvitationCommandMethod],
+		request:    request,
+	})
+
+	return respEnvelope
+}
+
+// AcceptInvitation accepts a stored connection invitation.
+func (de *DIDExchange) AcceptInvitation(request *models.RequestEnvelope) *models.ResponseEnvelope {
+	return exec(&restOperation{
+		url:        de.URL,
+		token:      de.Token,
+		httpClient: de.httpClient,
+		endpoint:   de.endpoints[cmddidexch.AcceptInvitationCommandMethod],
+		request:    request,
+	})
+}
+
+// CreateImplicitInvitation creates implicit invitation using inviter DID.
+func (de *DIDExchange) CreateImplicitInvitation(request *models.RequestEnvelope) *models.ResponseEnvelope {
+	return exec(&restOperation{
+		url:        de.URL,
+		token:      de.Token,
+		httpClient: de.httpClient,
+		endpoint:   de.endpoints[cmddidexch.CreateImplicitInvitationCommandMethod],
+		request:    request,
+	})
+}
+
+// AcceptExchangeRequest accepts a stored connection request.
+func (de *DIDExchange) AcceptExchangeRequest(request *models.RequestEnvelope) *models.ResponseEnvelope {
+	return exec(&restOperation{
+		url:        de.URL,
+		token:      de.Token,
+		httpClient: de.httpClient,
+		endpoint:   de.endpoints[cmddidexch.AcceptExchangeRequestCommandMethod],
+		request:    request,
+	})
+}
+
+// QueryConnections queries agent to agent connections.
+func (de *DIDExchange) QueryConnections(request *models.RequestEnvelope) *models.ResponseEnvelope {
+	return exec(&restOperation{
+		url:        de.URL,
+		token:      de.Token,
+		httpClient: de.httpClient,
+		endpoint:   de.endpoints[cmddidexch.QueryConnectionsCommandMethod],
+		request:    request,
+	})
+}
+
+// QueryConnectionByID fetches a single connection record by connection ID.
+func (de *DIDExchange) QueryConnectionByID(request *models.RequestEnvelope) *models.ResponseEnvelope {
+	return exec(&restOperation{
+		url:        de.URL,
+		token:      de.Token,
+		httpClient: de.httpClient,
+		endpoint:   de.endpoints[cmddidexch.QueryConnectionByIDCommandMethod],
+		request:    request,
+	})
+}
+
+// CreateConnection creates a new connection record in completed state and returns the generated connectionID.
+func (de *DIDExchange) CreateConnection(request *models.RequestEnvelope) *models.ResponseEnvelope {
+	return exec(&restOperation{
+		url:        de.URL,
+		token:      de.Token,
+		httpClient: de.httpClient,
+		endpoint:   de.endpoints[cmddidexch.CreateConnectionCommandMethod],
+		request:    request,
+	})
+}
+
+// RemoveConnection removes given connection record.
+func (de *DIDExchange) RemoveConnection(request *models.RequestEnvelope) *models.ResponseEnvelope {
+	return exec(&restOperation{
+		url:        de.URL,
+		token:      de.Token,
+		httpClient: de.httpClient,
+		endpoint:   de.endpoints[cmddidexch.RemoveConnectionCommandMethod],
+		request:    request,
+	})
+}
```

Comment:
Looks like duplicate code. These can be combined into one function and pass endpoint as an argument while creating restOperation.

Reply:
as suggested, i've added an unexported method that wraps the common lines